packages/@aws-cdk/aws-cognito-identitypool/lib/identitypool.ts

```diff
@@ -0,0 +1,547 @@
+   */
+  readonly appId: string
+}
+
+/**
+ * Login Provider for Identity Federation using Apple Credentials
+*/
+export interface IdentityPoolAppleLoginProvider {
+  /**
+   * App Id for Apple Identity Federation
+  */
+  readonly servicesId: string
+}
+
+/**
+ * Login Provider for Identity Federation using Google Credentials
+ */
+export interface IdentityPoolGoogleLoginProvider {
+  /**
+   * App Id for Google Identity Federation
+   */
+  readonly clientId: string
+}
+
+/**
+ * Login Provider for Identity Federation using Twitter Credentials
+ */
+export interface IdentityPoolTwitterLoginProvider {
+  /**
+   * App Id for Twitter Identity Federation
+   */
+  readonly consumerKey: string
+
+  /**
+   * App Secret for Twitter Identity Federation
+   */
+  readonly consumerSecret: string
+}
+
+/**
+ * Login Provider for Identity Federation using Digits Credentials
+ */
+export interface IdentityPoolDigitsLoginProvider extends IdentityPoolTwitterLoginProvider {}
+
+/**
+ * External Identity Providers To Connect to User Pools and Identity Pools
+ */
+export interface IdentityPoolProviders {
+
+  /** App Id for Facebook Identity Federation
+   * @default - No Facebook Authentication Provider used without OpenIdConnect or a User Pool
+   */
+  readonly facebook?: IdentityPoolFacebookLoginProvider
+
+  /** Client Id for Google Identity Federation
+   * @default - No Google Authentication Provider used without OpenIdConnect or a User Pool
+   */
+  readonly google?: IdentityPoolGoogleLoginProvider
+
+  /** App Id for Amazon Identity Federation
+   * @default -  No Amazon Authentication Provider used without OpenIdConnect or a User Pool
+   */
+  readonly amazon?: IdentityPoolAmazonLoginProvider
+
+  /** Services Id for Apple Identity Federation
+   * @default - No Apple Authentication Provider used without OpenIdConnect or a User Pool
+   */
+  readonly apple?: IdentityPoolAppleLoginProvider
+
+  /** Consumer Key and Secret for Twitter Identity Federation
+   * @default - No Twitter Authentication Provider used without OpenIdConnect or a User Pool
+   */
+  readonly twitter?: IdentityPoolTwitterLoginProvider
+
+  /** Consumer Key and Secret for Digits Identity Federation
+   * @default - No Digits Authentication Provider used without OpenIdConnect or a User Pool
+   */
+  readonly digits?: IdentityPoolDigitsLoginProvider
+}
+
+/**
+* Authentication providers for using in identity pool.
+* @see https://docs.aws.amazon.com/cognito/latest/developerguide/external-identity-providers.html
+*/
+export interface IdentityPoolAuthenticationProviders extends IdentityPoolProviders {
+
+  /**
+   * The User Pool Authentication Provider associated with this Identity Pool
+   * @default - no User Pools Associated
+   */
+  readonly userPool?: IUserPoolAuthenticationProvider;
+
+  /**
+   * The OpenIdConnect Provider associated with this Identity Pool
+   * @default - no OpenIdConnectProvider
+  */
+  readonly openIdConnectProvider?: IOpenIdConnectProvider;
```

Comment:
What if a user wants to provide multiple? Should we handle this the same as we do `userPool` (can configure one on creation and then multiple after the fact)?

Reply:
At first it seemed like even though this took an array, only one could be supported at the time, but I see now that multiple actually can be attached, so changing